src/Widgets/DeviceManagerWidget.vala

```diff
@@ -52,6 +54,29 @@ public class Sound.Widgets.DeviceManagerWidget : Gtk.Grid {
         update_showable ();
     }
 
+    private void disconnected () {
+        if (null == device_list) {
+            return;
+        }
+        foreach (unowned var child in device_list.get_children ()) {
+            device_list.remove (child);
+        }
+    }
+
+    private void update_device (Device device) {
+        if (device.input != is_input_manager) {
+            return;
+        }
+        if (null != device_list) {
+            foreach (unowned var child in device_list.get_children ()) {
+                if (device.display_name == ((DeviceItem) child).display_name) {
```

Comment:
Is there some sort of device ID that could be used here instead of the name? Although it could be rare, there could be an instance of someone connecting another device with the same name :)

Reply:
Great thought. I've adjusted this to use the device ID instead.